### ISSUE TITLE
New rule proposal: no-magic-timeouts

### ISSUE THREAD

**Issue by unlikelyzero** [mskelton/eslint-plugin-playwright#487]
## The problem

Playwright's auto-waiting means an explicit `timeout` is an escape hatch. In practice those escape hatches accumulate as bare numbers scattered across spec files:

```js
await page.goto('/reports', { timeout: 60000 })
await page.getByRole('button', { name: 'Generate' }).click({ timeout: 5000 })
await expect(page.getByText('Ready')).toBeVisible({ timeout: 30 * 1000 })
test.setTimeout(120000)
```

Each of those numbers records no reason. Nobody reading the file can tell whether `5000` was measured, guessed, or bumped in a hurry to get CI green — so nobody dares lower it, and the only safe edit is to raise it. Timeouts in a test suite therefore only ever ratchet upward.

Two concrete costs follow:

1. **Changes don't compose.** Because each literal lives at its own call site, "make the suite less flaky" turns into a hunt through every spec file. Identical values drift apart, and the suite ends up with no coherent timing policy — just a scatter of independently-tuned numbers.
2. **Regressions get masked.** The suite takes its effective upper bound from whichever number happens to be largest. A regression that used to fail in two seconds instead fails in sixty, or passes slowly and is never noticed at all.

Naming the value fixes both. `{ timeout: REPORT_GENERATION_TIMEOUT }` says why the wait exists, and editing the constant edits every site that shares that reason. Better still, most of these belong in `playwright.config.ts` as `timeout`, `expect.timeout`, `use.actionTimeout`, or `use.navigationTimeout`, where they apply suite-wide and can be overridden per project.

## Why this plugin rather than core `no-magic-numbers`

ESLint core's [`no-magic-numbers`](https://eslint.org/docs/latest/rules/no-magic-numbers) works by whitelisting *syntactic positions* — `VariableDeclarator`, object property, array index, default parameter — and it exempts object properties by default (`detectObjects: false`). Turning `detectObjects` on to catch `{ timeout: 5000 }` also flags every other numeric property in the file, so nobody enables it on a test suite.

That whitelist-by-position design is the part worth borrowing. Every shipped magic-value rule works this way, because judging literals in general produces unusable noise. A Playwright plugin is unusually well-placed to apply it: the Playwright API tells us exactly which positions are timeouts, so the rule can be precise where a general-purpose one cannot.

There is also useful precedent in the plugin already — `require-to-pass-timeout` reasons about the presence of a `timeout` option on `toPass()`. This proposal is about the *value* in that same position.

## Proposed rule

`no-magic-timeouts` — report numeric timeout values passed inline to a Playwright call.

Reported:

- any `timeout` property in an options object passed directly to a call: `click()`, `goto()`, `waitFor()`, `waitForURL()`, `toBeVisible()`, `toPass()`, the `test()` options object, `test.describe.configure()`
- `test.setTimeout(120000)` and `testInfo.setTimeout(120000)`
- arithmetic built purely from literals, e.g. `30 * 1000` and `2 * 60 * 1000`, which is how these values are usually written

Not reported, because the value already has a name or a home:

- a binding or member expression: `{ timeout: REPORT_TIMEOUT }`, `{ timeout: TIMEOUTS.short }`
- a named options object: `const uploadOptions = { timeout: 120_000 }`
- anything inside `defineConfig()`, and any object that isn't a call argument

Options:

- `allow: number[]` (default `[]`) — values permitted inline. Playwright treats `0` as "disable this timeout", which is self-documenting, so `{ allow: [0] }` is expected to be a common setting.
- `properties: string[]` (default `["timeout"]`) — property names treated as timeouts, so projects setting `actionTimeout` / `navigationTimeout` through `test.use()` can cover those too.

Opt-in rather than part of the recommended config: whether a given timeout deserves a name is a project-level judgement, and suites that already centralise their timeouts have nothing to fix.

## Detection

A `Property` visitor keyed on the property name, gated on the containing `ObjectExpression` being a direct call argument and not inside `defineConfig()`, plus a `CallExpression` visitor for `setTimeout`. Values are statically evaluated across `+ - * /` and unary `+`/`-` so arithmetic is covered, and evaluation returns nothing the moment it hits an identifier — a named reference being exactly what the rule is asking for.

Happy to open a PR with the implementation, tests, and docs if this sounds worth having.
